wordpress-mobile/WordPress-FluxC-Android
# Add support for users/%s/auth-options endpoint

### Description

This PR introduces an implementation for the `users/%s/auth-options` endpoint. 

This endpoint accepts either a username or an email address and, given it matches an existing account, a response with two boolean fields is returned:
- `passwordless`: indicates if the user has set a password for their account or not.
- `email_verified`: indicates if the account's associated email is verified or not.

If no matching account is found, a `unknown_user` error is returned instead.

We are mostly interested in the `passwordless` field, as it will allow us to further optimize our login flows on WPAndroid, specially the ones where we currently ask for a password even when the user haven't set one for their account.

### To test

You can either run `ReleaseStack_AccountTest` or use the following steps on the example app:
- On the initial screen, tap **Signed Out Actions**.
- On the next screen, tap **Fetch Auth Options**.
- On the dialog, enter an username or email address associated with an existing WP.com account and tap **Ok**.
- Check if the information printed in the console matches the account.

Feel free to try out different account combinations or check `ReleaseStack_AccountTest` for the most common scenarios.
